### PR TITLE
Fulfill Deployment Requirements

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This PR serves as a merge base for multiple upcoming, related changes regarding the deployment and packaging requirements of VAST. It's done this way because the way CMake must be ran to configure and build VAST will change multiple times over the course of this PR, and we don't want to constantly break master.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

n/t

### CODE REVIEW

File: libvast/vast/detail/tracepoint.hpp [hunk 1]
```diff
@@ -239,7 +239,7 @@
 
 
 #if defined(__ELF__) && (defined(__x86_64__) || defined(__i386__)) && \
-    !VAST_DISABLE_SDT
+    VAST_ENABLE_SDT
```
Comment (lava) [inline libvast/vast/detail/tracepoint.hpp:242]: Please don't forget to provide this macro by default if you flip the semantics here.

Comment (dominiklohmann) [reply libvast/vast/detail/tracepoint.hpp:242]: Already happening via `target_compile_definitions`

File: cmake/VASTMacDependencyPaths.cmake [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  # ports/brew/fink executables are not found.  If they are found, the actual
+  # paths will be pre-pended again below.
+  list(INSERT CMAKE_PREFIX_PATH 0 /opt/local)
+  list(INSERT CMAKE_PREFIX_PATH 0 /usr/local)
+  list(INSERT CMAKE_PREFIX_PATH 0 /sw)
+  if (APPLE AND "${PROJECT_SOURCE_DIR}" STREQUAL "${CMAKE_SOURCE_DIR}")
+    # Macports
+    find_program(MAC_PORTS_BIN ports)
+    if (MAC_PORTS_BIN)
+      list(INSERT CMAKE_PREFIX_PATH 0 ${MAC_PORTS_BIN})
```
Comment (tobim) [inline cmake/VASTMacDependencyPaths.cmake:16]:
```suggestion
      list(INSERT CMAKE_PREFIX_PATH 0 /opt/local)
```

File: cmake/VASTMacDependencyPaths.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# search prefix paths, but the nicer thing would be if they are prepended to the
+# default, so that is fixed here.
+if (NOT _MAC_DEPENDENCY_PATHS)
+  set(_MAC_DEPENDENCY_PATHS)
+  # Prepend the default search path locations, in case for some reason the
+  # ports/brew/fink executables are not found.  If they are found, the actual
+  # paths will be pre-pended again below.
+  list(INSERT CMAKE_PREFIX_PATH 0 /opt/local)
+  list(INSERT CMAKE_PREFIX_PATH 0 /usr/local)
+  list(INSERT CMAKE_PREFIX_PATH 0 /sw)
```
Comment (tobim) [inline cmake/VASTMacDependencyPaths.cmake:11]:
```suggestion
```

File: cmake/VASTMacDependencyPaths.cmake [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        COMMAND ${MAC_HOMEBREW_BIN} "--prefix"
+        OUTPUT_VARIABLE HOMEBREW_PREFIX
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+      list(INSERT CMAKE_PREFIX_PATH 0 ${HOMEBREW_PREFIX})
+      list(INSERT CMAKE_PREFIX_PATH 0 ${HOMEBREW_PREFIX}/opt/openssl)
+    endif ()
+    # fink
+    find_program(MAC_FINK_BIN fink)
+    if (MAC_FINK_BIN)
+      list(INSERT CMAKE_PREFIX_PATH 0 /sw) # Fink
```
Comment (tobim) [inline cmake/VASTMacDependencyPaths.cmake:31]:
```suggestion
      list(INSERT CMAKE_PREFIX_PATH 0 /sw)
```

**Review by tobim — CHANGES_REQUESTED** [review]

**Comment by tobim** [comment]
I'm running into a few errors:
```
CMake Error at CMakeLists.txt:88 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "INTERPROCEDURAL_OPTIMIZATION_RELEASE" is not allowed.


CMake Error at CMakeLists.txt:88 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_STANDARD" is not allowed.


CMake Error at CMakeLists.txt:88 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_STANDARD_REQUIRED" is not allowed.


CMake Error at CMakeLists.txt:88 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "CXX_EXTENSIONS" is not allowed.


CMake Error at CMakeLists.txt:234 (target_link_libraries):
  INTERFACE library can only be used with the INTERFACE keyword of
  target_link_libraries
```

**Review by tobim — APPROVED** [review]

**Comment by tobim** [comment]
I tested the current state locally and all my setups are working.

Note that it will be best to recreate the build directories. The caf and broker submodules must be initialized in their new locations under `libvast/aux/...`.

**Comment by dominiklohmann** [comment]
After the approval I only squashed some PRs in the middle.

**Pull request merged**